jetty-http/src/main/java/org/eclipse/jetty/http/compression/HuffmanDecoder.java

```diff
@@ -51,17 +61,19 @@ public String decode(ByteBuffer buffer) throws EncodingException
             while (_bits >= 8)
             {
                 int c = (_current >>> (_bits - 8)) & 0xFF;
-                _node = tree[_node * 256 + c];
+                _node = Huffman.tree[_node * 256 + c];
                 if (rowbits[_node] != 0)
                 {
-                    if (rowsym[_node] == EOS)
+                    if (rowsym[_node] == Huffman.EOS)
                     {
                         reset();
                         throw new EncodingException("eos_in_content");
                     }
 
                     // terminal node
-                    _utf8.append((byte)(0xFF & rowsym[_node]));
+                    int i = 0xFF & rowsym[_node];
+                    i = Huffman.getReplacementCharacter(i);
```

Comment:
I don't understand why this is here? Why replace here?  Don't we only replace when converting characters to bytes?

Also the method is poorly named, as it only replaces bad characters.  It is more a sanitise.    But even then, that is applying a HTTP sanity check that perhaps should not be in the base Huffman code?

Reply:
I have it now so we won't encode the illegal vchars, but when we decode we replace any bad vchars.

So here someone could send us an encoded huffman string with control chars etc, and we don't want to leave these in the decoded field value.